src/acvp_rsa_sig.c

```diff
@@ -399,26 +399,42 @@ static ACVP_RESULT acvp_rsa_sig_kat_handler_internal(ACVP_CTX *ctx, JSON_Object
 
 
             if (alg_id == ACVP_RSA_SIGVER) {
-                signature = (char *)json_object_get_string(testobj, "signature");
-                if (!signature) {
+                tmp_signature = (char *)json_object_get_string(testobj, "signature");
+                if (!tmp_signature) {
                     ACVP_LOG_ERR("Missing 'signature' from server json");
                     rv = ACVP_MISSING_ARG;
                     json_value_free(r_tval);
                     goto err;
                 }
-                json_siglen = strnlen_s(signature, ACVP_RSA_SIGNATURE_MAX + 1);
+                json_siglen = strnlen_s(tmp_signature, ACVP_RSA_SIGNATURE_MAX + 1);
                 if (json_siglen > ACVP_RSA_SIGNATURE_MAX) {
                     ACVP_LOG_ERR("'signature' too long in server json");
                     rv = ACVP_INVALID_ARG;
                     json_value_free(r_tval);
                     goto err;
                 }
+
+                if (json_siglen != mod/4) {
+                    signature = calloc(mod/4 +1, sizeof(char));                    
```

Comment:
Need to check is calloc was successful here?

Reply:
yep.